app/store/middleware.js

```diff
@@ -143,30 +148,32 @@ function cleanupState(action, keepCurrent = false) {
 
     postIdsToKeep.forEach((postId) => {
         const post = payload.entities.posts.posts[postId];
-        const skip = keepCurrent && currentChannelId === post.channel_id;
-
-        if (!skip && retentionPeriod && (Date.now() - post.create_at) / (1000 * 3600 * 24) > retentionPeriod) {
-            const postsInChannel = nextEntitites.posts.postsInChannel[post.channel_id];
-            const index = postsInChannel.indexOf(postId);
-            if (index !== -1) {
-                postsInChannel.splice(index, 1);
+        if (post) {
```

Comment:
If we have the post id, but not the post, will that cause any problems later on?

Reply:
we already take care of that [here](https://github.com/mattermost/mattermost-mobile/pull/1075/files#diff-60e3906018bf9f32d47eeea2c937dc0dR155)

Reply:
But then I have to fix that cause that code is unreachable right now in that case, will submit a fix soon